src/lap_stats/lap_stats.c

```diff
@@ -247,54 +357,39 @@ static void onLocationUpdated(const GpsSnapshot *gpsSnapshot) {
    static int sectorEnabled = 0;
    static int startFinishEnabled = 0;
 
-   g_distance += calcDistancesSinceLastSample(gpsSnapshot);
-
-   LoggerConfig *config = getWorkingLoggerConfig();
+   const LoggerConfig *config = getWorkingLoggerConfig();
    const GeoPoint *gp = &gpsSnapshot->sample.point;
    const float targetRadius = degreesToMeters(config->TrackConfigs.radius);
 
+   // FIXME.  This active track configuration here is le no good.
```

Comment:
Plainly put the code should not be here.  This code should be invoked right after a configuration has been changed on the track settings.